src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -136,6 +139,49 @@ public void Dispose()
         public DataFrame Table(string tableName) =>
             new DataFrame((JvmObjectReference)_jvmObject.Invoke("table", tableName));
 
+        /// <summary>
+        /// Creates a `DataFrame` from an IEnumerable containing [[GenericRow]]s using the given schema.
+        /// It is important to make sure that the structure of every [[GenericRow]] of the provided IEnumerable matches
+        /// the provided schema. Otherwise, there will be runtime exception.
+        /// </summary>
+        /// <param name="data">List of Row objects</param>
+        /// <param name="schema">Schema as StructType</param>
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data, StructType schema) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.sql.api.dotnet.SQLUtils",
+                "createDataFrameHelper",
+                this,
+                data,      
+                DataType.FromJson(_jvmObject.Jvm, schema.Json)));
+
+        /// <summary>
+        /// Creates a dataframe from an IEnumerable containing [[GenericRow]]s by inferring the schema from the given data.
+        /// </summary>
+        /// <param name="data">List of Row objects</param>        
+        /// <returns>DataFrame object</returns>
+        public DataFrame CreateDataFrame(IEnumerable<GenericRow> data)
```

Comment:
I don't see the value of making this as public when the functionalities are very limited. Why not expose `CreateDataFrame(IEnumerable<string> data)` and `CreateDataFrame(IEnumerable<int> data)` instead? It's much easier to use that way.

Reply:
This is currently doing something similar to how pyspark is inferring the schema.  Although how useful this API is for users, I'm not sure.  If the simpler API you suggest (limited to a single column of one type) is more useful, then it's worth exploring.  @Niharikadutta what do you think ?

**Examples:**
pyspark:
```python
>>> spark.createDataFrame([["hello"], ["world"]])
DataFrame[_1: string]
>>> spark.createDataFrame([["hello", 1], ["world", 2]])
DataFrame[_1: string, _2: bigint]
>>> spark.createDataFrame(["hello", "world"], "string")
DataFrame[value: string]
>>> spark.createDataFrame(["hello", "world"], StringType())
DataFrame[value: string]
>>> spark.createDataFrame(["hello", "world"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\bin\spark\spark-2.4.3-bin-hadoop2.7\python\pyspark\sql\session.py", line 748, in createDataFrame
    rdd, schema = self._createFromLocal(map(prepare, data), schema)
  File "C:\bin\spark\spark-2.4.3-bin-hadoop2.7\python\pyspark\sql\session.py", line 416, in _createFromLocal
    struct = self._inferSchemaFromList(data, names=schema)
  File "C:\bin\spark\spark-2.4.3-bin-hadoop2.7\python\pyspark\sql\session.py", line 348, in _inferSchemaFromList
    schema = reduce(_merge_type, (_infer_schema(row, names) for row in data))
  File "C:\bin\spark\spark-2.4.3-bin-hadoop2.7\python\pyspark\sql\session.py", line 348, in <genexpr>
    schema = reduce(_merge_type, (_infer_schema(row, names) for row in data))
  File "C:\bin\spark\spark-2.4.3-bin-hadoop2.7\python\pyspark\sql\types.py", line 1062, in _infer_schema
    raise TypeError("Can not infer schema for type: %s" % type(row))
TypeError: Can not infer schema for type: <class 'str'>
```

spark-shell:
```scala
scala> spark.createDataFrame(Seq(Tuple1("hello"), Tuple1("world")))
res17: org.apache.spark.sql.DataFrame = [_1: string]
scala> spark.createDataFrame(Seq(("hello", 1), ("world", 2)))
res18: org.apache.spark.sql.DataFrame = [_1: string, _2: int]
```

Reply:
We can do the inferring later as a separate PR if there is a demand (and we can always use the API where you can specify the schema).

Reply:
@imback82 Sure I can change that to have separate APIs for different types. Do you want me to expose one for each of the following: string, int, float, double, etc? Or just string and int for now?